File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        even=None
        even_tail=None
        odd=None
        odd_tail=None

        temp=head
        index=0
        while(temp!=None):
            if(index%2==0):
                if even==None:
                    even=temp
                    even_tail=temp
                else:
                    even_tail.next=temp
                    even_tail=even_tail.next
            
            else:
                if odd==None:
                    odd=temp
                    odd_tail=temp
                else:
                    odd_tail.next=temp
                    odd_tail=odd_tail.next
            index+=1
            temp=temp.next
        
        if even_tail:
            even_tail.next=odd
        else:
            even=odd

        if odd_tail:   
            odd_tail.next=None

        return even

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if(head==null || head.next==null)
            return head;

        ListNode left = head;
        ListNode right = head.next;

        ListNode leftTemp = head;
        ListNode rightTemp = head.next;
        while(rightTemp!=null && rightTemp.next!=null){
            leftTemp.next = leftTemp.next.next;
            leftTemp=leftTemp.next;

            rightTemp.next = rightTemp.next.next;
            rightTemp=rightTemp.next;
        }

        leftTemp.next=right;
        return left;

    }
}